sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java

```diff
@@ -164,21 +239,42 @@ private HttpResponse attempt(final HttpRequest httpRequest, final HttpPipelineNe
 
             return attempt(httpRequest, next, tryCount + 1, suppressed);
         } else {
-            if (tryCount >= retryStrategy.getMaxRetries()) {
+            if (tryCount >= maxRetries) {
                 logRetryExhausted(tryCount);
             }
 
             return httpResponse;
         }
     }
 
-    private static boolean shouldRetry(RetryStrategy retryStrategy, HttpResponse response, int tryCount) {
-        return tryCount < retryStrategy.getMaxRetries() && retryStrategy.shouldRetry(response);
+    /*
+     * Determines the delay duration that should be waited before retrying.
+     */
+    private static Duration determineDelayDuration(HttpResponse response, int tryCount, RetryStrategy retryStrategy,
+                                                   HeaderName retryAfterHeader, ChronoUnit retryAfterTimeUnit) {
+        // If the retry after header hasn't been configured, attempt to look up the well-known headers.
+        if (retryAfterHeader == null) {
+            return getWellKnownRetryDelay(response.getHeaders(), tryCount, retryStrategy, OffsetDateTime::now);
+        }
+
+        String retryHeaderValue = response.getHeaderValue(retryAfterHeader);
+
+        // Retry header is missing or empty, return the default delay duration.
+        if (isNullOrEmpty(retryHeaderValue)) {
+            return retryStrategy.calculateRetryDelay(tryCount);
+        }
+
+        // Use the response delay duration, the server returned it for a reason.
+        return Duration.of(Integer.parseInt(retryHeaderValue), retryAfterTimeUnit);
```

Comment:
Pending comments about keeping retryAfterHeader logic, do we want Integer.parseInt to fail here if the retryHeaderValue isn't a number?